system/lib/wasmfs/backends/opfs_backend.cpp

```diff
@@ -38,6 +57,9 @@ class Worker {
       func();
     }
   }
+
+  em_promise_t getPromise() { return promise; }
```

Comment:
Do we need this API defined in the sync case? It wouldn't be used I assume.

(Or, if we must have it, could it be null rather than create a not-really-used promise?)

Reply:
This is for the non-threads implementation of `Worker`, i.e. the one used when JSPI is enabled without threads. We definitely want to be able to call `wasmfs_create_opfs_backend_async` on the main thread in that configuration even though `wasmfs_create_opfs_backend` is also valid to call, just so that there is some function that is always safe to call from any thread and in any configuration. Providing a uniform interface here, at the point where we decide whether to use a worker thread or not, simplifies everything else.

Reply:
But does that uniform interface need `getPromise` in the path where promises are not used? IIUC in the code, the promise in that case is not really used, which seems odd.